sylt-compiler/src/typechecker.rs

```diff
@@ -1475,6 +1483,50 @@ impl TypeChecker {
                         var
                     ),
                 },
+
+                Constraint::TotalEnum(vars) => match self.find_type(a) {
+                    Type::Unknown => Ok(()),
+
+                    Type::Enum(enum_name, enum_vars) => {
+                        let mut missing = Vec::new();
+                        for var in vars.iter() {
+                            if !enum_vars.contains_key(var) {
+                                missing.push(var.clone());
+                            }
+                        }
+                        if !missing.is_empty() {
+                            err_type_error!(
```

Comment:
Return the errors and flatten out the if-statements?

Reply:
We cannot do this, since we add help to all errors in unify. We cannot use return.